server/app/models.py

```diff
@@ -234,6 +234,13 @@ def get_course_info(self, course):
                 assign_info['final']['backup'] = \
```

Comment:
All these long lines would go away if this block started with 

`final = assign_info['final']`